### PR TITLE
feat/add-doc-previewer

### LINKED ISSUE #21: Preview OpenRPC command
**Describe the feature you would like**
I'd like to preview my `OpenRPC` Document via the `vscode` extension.

**Describe the solution**
I'd like to use https://github.com/open-rpc/docs-react in some way to render the current editor text as `Documentation`. 

**Additional Notes**
here is an example of one way to do it: https://github.com/arjun-g/vs-swagger-viewer

### PR DESCRIPTION
**Pull request by vijya76**
This PR adds the feature to preview the doc inside a webview.

Currently, the extension will search for `openrpc.json` in the root of workspace and open that in the preview panel.

Also, the preview gets updated when the document is saved.

### How to use
- `ctrl + shift + p`
- search for `Start OpenRPC preview window`

fixes #21

### CODE REVIEW

**Comment by shanejonas** [comment]
@vijya76 this is awesome!!

I played around with getting current documents text. should we go that route? The extension is already only enabled for `*openrpc.json` in the `package.json` so we know it's the right one.

```
const { document } = vscode.window.activeTextEditor;       
const documentString = document.getText();
```

I got it up and running already. looks awesome, works great.

image


Have you had any thoughts on how to deal with columns? (`vscode.ViewColumn.One`), I've seen 1-2 extensions just have 2 commands, one called like `Split OpenRPC Preview`(`vscode.ViewColumn.Two`) and `OpenRPC Preview` (`vscode.ViewColumn.One`). Wonder what other people are doing, personally I'd want it to split always. But I can see how that might not be the best default.

File: extension/ReactWebview.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+			vscode.workspace.findFiles('openrpc.json')
+				.then(files => {
+					const openRpcDoc = files[0]; // using the file at root
+
+					const docData = require(openRpcDoc.path);
+					this.updateContent(docData);
+				});
+
+			vscode.workspace.onDidChangeTextDocument(({ contentChanges, document }) => {
+				if (document.fileName.endsWith('openrpc.json') && !document.isDirty) {
```
Comment (shanejonas) [inline extension/ReactWebview.ts]:
```suggestion
				if (!document.isDirty) {
```

Comment (shanejonas) [reply extension/ReactWebview.ts]: this gets taken care of in the [package.json#L25](https://github.com/open-rpc/editor-extensions-vscode/blob/master/package.json#L25)

**Comment by devonwesley** [comment]
Nice, man good stuff. 

I don't know if it's my editor thats messing up the re-rendering, but when I have both of the windows open, `openrpc.json` and the `html-view`, this happens:

- I'm in the `html-view` // `It's working fine`
- and click back to `openrpc.json`
- then back again to the `html-view` // `not working fine`

The `html-view` doesn't re-render for me, when I click back to the `html-view` window.

Works perfectly fine first time around, then something gets upset with I click back and forth.

**Comment by shanejonas** [comment]
re-opened this at #54

**Pull request closed without merging**